presto-spi/src/main/java/com/facebook/presto/spi/relation/ExistsRowExpression.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.relation;
+
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.spi.SourceLocation;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.facebook.presto.common.type.BooleanType.BOOLEAN;
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * RowExpression equivalent of ExistsPredicate.
+ */
+public class ExistsRowExpression
+        extends IntermediateFormRowExpression
+{
+    private final RowExpression subquery;
```

Comment:
I think Exists have a PlanNode as a child and not a (scalar) RowExpression?

Reply:
Discussed offline